### PR TITLE
Add topics management

### PR DESCRIPTION
**Pull request by taktran**
Add the ability to list topics, add topics and remove topics from repositories.

## Usage

```bash
./bin/asari.js repos list-topics <github-url>

# List all topics.
```

```bash
./bin/asari.js repos <add-topics|remove-topics> <github-url> --topic new-app

# Add/Remove a topic

./bin/asari.js repos <add-topics|remove-topics> <github-url> --topic new-app --topic good-one
./bin/asari.js repos <add-topics|remove-topics> <github-url> --topics new-app,good-one
./bin/asari.js repos <add-topics|remove-topics> <github-url> --topics "new-app,  good-one"

# Add/Remove multiple topics
```

## Reference

For doing this https://github.com/Financial-Times/next/issues/386

### CODE REVIEW

**Comment by sjparkinson** [comment]
I'd suggest that we add any topics command as a subcommand under a `repo` command.

That way the main commands of this CLI line up with the high level resources listed by the v3 GitHub API, i.e. `issues`, `repos`, `projects`, `pulls`, `apps`, `gists`, etc.

File: README.md [hunk 1]
```diff
@@ -182,6 +185,40 @@ npx asari pulls open <github-url>
...
+```
+
+```bash
+npx asari topics add <github-url> --topic new-app
+
+# Add a topic
+
+npx asari topics add <github-url> --topic new-app --topic good-one
+npx asari topics add <github-url> --topic new-app,good-one
+npx asari topics add <github-url> --topic "new-app,  good-one"
```
Comment (sjparkinson) [inline README.md]: I don't think this example is clear about if it's adding a single topic or two.

File: README.md [hunk 2]
```diff
@@ -182,6 +185,40 @@ npx asari pulls open <github-url>
...
+### Working with GitHub Topics
+
+```bash
+npx asari topics list <github-url>
+
+# List all topics.
+```
+
+```bash
+npx asari topics add <github-url> --topic new-app
```
Comment (sjparkinson) [inline README.md]: Could trim this to:

```suggestion
npx asari topics <add|remove> <github-url> --topic new-app
```

**Comment by taktran** [comment]
> I'd suggest that we add any `topics` command as a subcommand under a `repo` command.

Good point :+1:

File: src/commands/topics/add.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	return baseOptions(yargs).example('github-url', 'Pattern: https://github.com/[owner]/[repository]')
+}
+
+/**
+ * Add topics to a repository.
+ *
+ * @param {object} argv - argv parsed and filtered by yargs
+ * @param {string} argv.token
+ * @param {string} argv.json
+ * @param {object} argv.githubUrl - The GitHub url parsed in the withGitHubUrl() yarg option into appropriate properties, such as `owner` and `repo`.
```
Comment (taktran) [inline src/commands/topics/add.js]: Add doc about `topic` param

File: src/lib/common-yargs.js [hunk 4]
```diff
@@ -179,6 +179,20 @@ const withGitHubUrl = options => yargs => {
...
 }
 
+const withTopic = options => yargs => {
+	return yargs.option(
+		'topic',
+		Object.assign(
+			{
+				alias: ['topics'],
+				type: 'string',
+				describe: 'GitHub topic to add. To add more than one, add multiple option and values or a comma separated list',
```
Comment (taktran) [inline src/lib/common-yargs.js]: Can put comma separated logic here

File: test/commands/topics/list.test.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+beforeEach(() => {
+	jest
+		.spyOn(console, 'log')
+		.mockImplementation()
+		.mockName('console.log')
+})
+
+afterEach(() => {
+	jest.resetAllMocks()
```
Comment (taktran) [inline test/commands/topics/list.test.js]: Needs a `nock.cleanAll()`

File: README.md (hunk 2 above)
Comment (taktran) [reply README.md]: Also for moving to `repos` subcommand, we would have `repos add-topic` and `repos remove-topic`

**Comment by taktran** [comment]
@adambraimbridge @sjparkinson - have updated with the changes to using `repos` subcommand and other general tidy ups. Good for another review 👀

**Comment by taktran** [comment]
Partially does this: https://github.com/Financial-Times/asari/issues/30

**Pull request merged**